src/server.cpp

```diff
@@ -1191,60 +1204,112 @@ void CServer::MixEncodeTransmitData ( const int iChanCnt,
     else
     {
         // Stereo target channel -----------------------------------------------
+
+        const int maxPanDelay = MAX_DELAY_PANNING_SAMPLES;
+
+        int iLpan, iRpan, iPan;
+
         for ( j = 0; j < iNumClients; j++ )
         {
             // get a reference to the audio data and gain/pan of the current client
             const CVector<int16_t>& vecsData = vecvecsData[j];
+            const CVector<int16_t>& vecsData2 = vecvecsData2[j];
+
             const float             fGain    = vecvecfGains[iChanCnt][j];
-            const float             fPan     = vecvecfPannings[iChanCnt][j];
+            const float             fPan     = bDelayPan ? 0.5f : vecvecfPannings[iChanCnt][j];
+            const int iPanDel = lround( (float)( 2 * maxPanDelay - 2 ) * ( vecvecfPannings[iChanCnt][j] - 0.5f ) );
+            const int iPanDelL = ( iPanDel > 0 ) ? iPanDel : 0;
+            const int iPanDelR = ( iPanDel < 0 ) ? -iPanDel : 0;
 
             // calculate combined gain/pan for each stereo channel where we define
             // the panning that center equals full gain for both channels
             const float fGainL = MathUtils::GetLeftPan ( fPan, false ) * fGain;
             const float fGainR = MathUtils::GetRightPan ( fPan, false ) * fGain;
 
-            // if channel gain is 1, avoid multiplication for speed optimization
-            if ( ( fGainL == 1.0f ) && ( fGainR == 1.0f ) )
```

Comment:
This code seems to get lost and was annotated to be a speed optimization. I can't tell how important it was. Saving some multiplications sounds really minor (if the cost is two additional equality checks), but as this is part of the performance-critical code path we may want to double-check.